### PR TITLE
PROGMEM methods for Print.cpp

### PR DESCRIPTION
**Pull request by tius2000**
* missing PROGMEM methods for class Print added
* dynamic stack allocation for _printf_ moved to new method _vprintf_ 

However, I have no idea how to fix the conflict with the new macro _printf_P_ in FakePgmSpace.h :-(

(see #1067)

### CODE REVIEW

File: Sming/Wiring/Print.cpp [hunk 1]
```diff
@@ -197,31 +198,63 @@ size_t Print::println(const Printable &p)
...
-			{
-				write(tempBuff,sz);
-			}
+	size_t size = INITIAL_PRINTF_BUFFSIZE - 1;
+	
+	//	need to retry until size is big enough as m_vsnprintf is currently broken
+	//	(returns maxLen instead of required size)
+	while (1) {
+		char buffer[size + 1];
+		auto sz = m_vsnprintf(buffer, sizeof(buffer), fmt, va);
```
Comment (ADiea) [inline Sming/Wiring/Print.cpp]: > I guess this PR should wait for #1077. Then you shuld have directly the correct size needed +1 ending char. So I would not use while(1) and wait for the PR

File: Sming/Wiring/Print.cpp [hunk 2]
```diff
@@ -197,31 +198,63 @@ size_t Print::println(const Printable &p)
...
-	} while (retry);
+		size = sz;	
+	}
+}
+
+size_t Print::printf(const char *fmt, ...)
+{
+	va_list va;
+	va_start(va, fmt);
+	auto cnt = vprintf(fmt, va);
```
Comment (ADiea) [inline Sming/Wiring/Print.cpp]: Please only use auto when hiding an ugly/templetized type specifier. In this case it is size_t. IMHO it's faster to read and debug if you know the types.

File: Sming/Wiring/Print.cpp (hunk 1 above)
Comment (tius2000) [inline Sming/Wiring/Print.cpp]: You are right @ADiea. I just added automatic PROGMEM support to #1077, so this PR might even no longer be needed.

File: Sming/Wiring/Print.cpp (hunk 2 above)
Comment (tius2000) [inline Sming/Wiring/Print.cpp]: ok

File: Sming/Wiring/Print.h [hunk 3]
```diff
@@ -26,12 +26,14 @@
 
 #define INITIAL_PRINTF_BUFFSIZE 128
 
+#undef printf_P
```
Comment (ADiea) [inline Sming/Wiring/Print.h]: You should remove this here. If you do not want name collision you have 2 options: namespaces or explicit calling e.g. this.printf_P

Comment (tius2000) [inline Sming/Wiring/Print.h]: You are right, this is ugly and will cause problems. Therefore I would suggest the following procedure:

* wait for PR #1077 
* optional: implement _printf_P()_ etc. as functions instead a macros in FakePgmSpace (can be changed to an inline wrapper for _printf()_ if #1077 is accepted)
* remove _printf_P()_ from this PR as it is no longer needed
* add printf_P() and println_P() instead

Comment (ADiea) [inline Sming/Wiring/Print.h]: Yes, I guess if printf / m_printf will be progmem safe, then no need for _P variant. For printf I do not care about speed so much, but sprintf I would keep it fast

**Pull request closed without merging**